File: benchmarks/solutions/kotlin/a1-two-sum.kt
fun twoSum(nums: IntArray, target: Int): Pair<Int, Int> {
    val seen = mutableMapOf<Int, Int>()
    for ((i, n) in nums.withIndex()) {
        val complement = target - n
        seen[complement]?.let { return it to i }
        seen[n] = i
    }
    error("no solution")
}

File: benchmarks/solutions/erlang/a1-two-sum.erl
-module(two_sum).
-export([two_sum/2]).

two_sum(Nums, Target) ->
    two_sum(Nums, Target, #{}, 0).

two_sum([], _Target, _Seen, _I) ->
    {-1, -1};
two_sum([N | Rest], Target, Seen, I) ->
    Complement = Target - N,
    case maps:find(Complement, Seen) of
        {ok, J} -> {J, I};
        error -> two_sum(Rest, Target, Seen#{N => I}, I + 1)
    end.
